ironfish-cli/src/commands/evm/shield.ts

```diff
@@ -0,0 +1,122 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { GoldTokenJson } from '@ironfish/ironfish-contracts'
+import { ContractArtifact, EthUtils, GLOBAL_CONTRACT_ADDRESS } from '@ironfish/sdk'
+import { Flags, ux } from '@oclif/core'
+import { ethers } from 'ethers'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+
+export class ShieldCommand extends IronfishCommand {
+  static description = `Shield asset from EVM to private token`
+
+  static flags = {
+    ...LocalFlags,
+    amount: Flags.integer({
+      char: 'a',
+      description: 'The amount of the asset to shield',
+    }),
+    contractAddress: Flags.string({
+      char: 'c',
+      description: 'The EVM contract address of the asset to shield',
+    }),
+    nonce: Flags.integer({
+      char: 'n',
+      description: 'The nonce of the EVM transaction',
+    }),
+    to: Flags.string({
+      char: 't',
+      description: 'The Ironfish public address of the recipient',
+    }),
+    from: Flags.string({
+      char: 'f',
+      description: 'The account name of the sender',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(ShieldCommand)
+    let { amount, contractAddress, nonce, fee, to, privateKey, from } = flags
+    const client = await this.sdk.connectRpc()
+
+    if (!amount) {
+      const input = await ux.prompt('Enter amount to send: ', {
+        required: true,
+      })
+      amount = parseInt(input)
+    }
+
+    if (!nonce) {
```

Comment:
I think this can be optional since `eth/sendTransaction` will use the nonce of the account by default